src/crypto/key_expansion.rs

```diff
@@ -5,22 +5,58 @@ use super::{cipher_suite::CipherSuite, secret::Secret};
 use crate::error::Result;
 
 pub trait KeyExpansion {
-    fn expand_from<T>(cipher_suite: &CipherSuite, key_material: T) -> Result<Secret>
+    fn expand_from<M, K>(cipher_suite: &CipherSuite, key_material: M, key_id: K) -> Result<Secret>
     where
-        T: AsRef<[u8]>;
+        M: AsRef<[u8]>,
+        K: Into<u64>;
 }
 
-pub const SFRAME_HKDF_SALT: &[u8] = b"SFrame10";
-pub const SFRAME_HKDF_KEY_EXPAND_INFO: &[u8] = b"key";
-pub const SFRAME_HDKF_SALT_EXPAND_INFO: &[u8] = b"salt";
+pub fn get_hkdf_key_expand_info(key_id: u64) -> Vec<u8> {
+    [
+        SFRAME_LABEL,
+        SFRAME_HKDF_KEY_EXPAND_INFO,
+        &key_id.to_be_bytes(),
+    ]
+    .concat()
+}
 
-#[cfg(feature = "openssl")]
-pub const SFRAME_HKDF_SUB_SALT: &[u8] = b"SFrame10 AES CTR AEAD";
-#[cfg(feature = "openssl")]
-pub const SFRAME_HKDF_SUB_ENC_EXPAND_INFO: &[u8] = b"enc";
-#[cfg(feature = "openssl")]
-pub const SFRAME_HDKF_SUB_AUTH_EXPAND_INFO: &[u8] = b"auth";
+pub fn get_hkdf_salt_expand_info(key_id: u64) -> Vec<u8> {
+    [
+        SFRAME_LABEL,
+        SFRAME_HDKF_SALT_EXPAND_INFO,
+        &key_id.to_be_bytes(),
+    ]
+    .concat()
+}
+
+const SFRAME_LABEL: &[u8] = b"SFrame 1.0 ";
+
+// For the current test vectors different labels than specified were used
+// see https://github.com/sframe-wg/sframe/issues/137
+cfg_if::cfg_if! {
```

Comment:
https://doc.rust-lang.org/beta/core/macro.cfg_match.html#example

Reply:
Unfortunately this is still nightly: https://github.com/rust-lang/rust/issues/115585